tools/webtransport/h3/webtransport_h3_server.py

```diff
@@ -0,0 +1,247 @@
+
+        status = any(
+            header[0] == b":status" for header in response_headers)
+        if not status:
+            response_headers.append((b":status", b"200"))
+        self._http.send_headers(
+            stream_id=stream_id, headers=response_headers)
+
+        self._handler.session_established()
+
+    def _create_event_handler(self, session_id: int, path: bytes, request_headers: Dict[bytes, bytes]):
+        parsed = urlparse(path.decode())
+        file_path = os.path.join(handlers_path, parsed.path.lstrip('/'))
+        callbacks = {"__file__": file_path}
+        with open(file_path) as f:
+            exec(compile(f.read(), path, "exec"), callbacks)
+        session = WebTransportSession(self, session_id, request_headers)
+        return EventHandler(session, callbacks)
+
+
+class WebTransportSession(object):
+    """
+    A WebTransport session.
+    """
+
+    def __init__(self, protocol: WebTransportH3Protocol, session_id: int, request_headers: Dict[bytes, bytes]) -> None:
+        self.session_id = session_id
+        self.request_headers = request_headers
+
+        self._protocol: WebTransportH3Protocol = protocol
+        self._http: H3Connection = protocol._http
+
+    def stream_is_unidirectional(self, stream_id: int) -> bool:
+        """Returns True if the stream is unidirectional."""
+        return stream_is_unidirectional(stream_id)
+
+    def close(self, error_code: int = QuicErrorCode.NO_ERROR) -> None:
+        """
+        Close the session.
+
+        :param error_code: An error code indicating why the session is
+                           being closed.
+        """
+        self._http._quic.close(error_code=error_code)
+
+    def create_unidirectional_stream(self) -> int:
+        """
+        Create a unidirectional WebTransport stream and return the stream ID.
+        """
+        return self._http.create_webtransport_stream(session_id=self.session_id, is_unidirectional=True)
+
+    def create_bidirectional_stream(self) -> int:
+        """
+        Create a bidirectional WebTransport stream and return the stream ID.
+        """
+        stream_id = self._http.create_webtransport_stream(
+            session_id=self.session_id, is_unidirectional=False)
+        return stream_id
+
+    def send_stream_data(self, stream_id: int, data: bytes, end_stream: bool = False) -> None:
+        """
+        Send data on the specific stream.
+
+        :param stream_id: The stream ID on which to send the data.
+        :param data: The data to send.
+        :param end_stream: If set to True, the stream will be closed.
+        """
+        self._http._quic.send_stream_data(
+            stream_id=stream_id, data=data, end_stream=end_stream)
+
+    def send_datagram(self, data: bytes) -> None:
+        """
+        Send data using a datagram frame.
+
+        :param data: The data to send.
+        """
+        self._http.send_datagram(flow_id=self.session_id, data=data)
+
+
+class EventHandler(object):
```

Comment:
You don't need to explicitly inherit from `object` in py3.

Reply:
Removed.